.github/workflows/release.yml

```diff
@@ -18,21 +20,41 @@ jobs:
         with:
           java-version: 1.8
 
-      - name: Perform Maven Release
-        uses: qcastel/github-actions-maven-release@master
+      - name: Configure git
+        run: |
+          git config --global user.email "infrastructure+deploy@renttherunway.com"
+          git config --global user.name "rtr-deploy"
+
+      - name: Setup GPG
+        env:
+          GPG_KEY: ${{ secrets.GPG_KEY }}
+        run: |
+          echo "$GPG_KEY" | base64 -d > private.key
+          gpg --import --no-tty --batch --yes ./private.key
+          rm ./private.key
+
+      - name: Setup Nexus authentication
+        uses: s4u/maven-settings-action@v2.1.0
         with:
-          release-branch-name: "master"
-
-          git-release-bot-name: "rtr-actions"
-          git-release-bot-email: "infrastructure+github@renttherunway.com"
-
-          gpg-enabled: "true"
-          gpg-key-id: ${{ secrets.GPG_KEY_ID }}
-          gpg-key: ${{ secrets.GPG_KEY }}
-
-          maven-repo-server-username: ${{ secrets.OSS_NEXUS_USERNAME }}
-          maven-repo-server-password: ${{ secrets.OSS_NEXUS_PASSWORD }}
+          servers: '[
+          {
+            "id": "sonatype-nexus-staging",
+            "username": "${{ secrets.OSS_NEXUS_USERNAME }}",
+            "password": "${{ secrets.OSS_NEXUS_PASSWORD }}"
+          },
+          {
+            "id": "sonatype-nexus-snapshots",
+            "username": "${{ secrets.OSS_NEXUS_USERNAME }}",
+            "password": "${{ secrets.OSS_NEXUS_PASSWORD }}"
+          }]'
+
+      - name: Setup Github SSH key
+        uses: webfactory/ssh-agent@v0.4.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_KEY }}
 
-          maven-args: "--no-transfer-progress"
+      - name: Prepare release
```

Comment:
Very nice!  What happens if there is a failure, can we add a `mvn release:rollback` step for good measure if there's failure, so we don't leave it in a half-released state?